### ISSUE TITLE
missleaging light sectors

### ISSUE THREAD

**Issue by quantenschaum** [wellenvogel/ochartsng#13]
I stumbled upon the following: The light Travemünde, N-Mole is shown on the map to have a white sector to NE, but this white sector is actually a fog light. In normal operation this light has one red and one green sector. IMHO this is misleading and potentially very dangerous.

![ochartsng](https://github.com/wellenvogel/ochartsng/assets/3985599/83f0ee69-360f-4ee7-9505-49dd2e4cb4e4)

compare to http://waddenzee.duckdns.org/#17/53.96171/10.88879

![QMAP](https://github.com/wellenvogel/ochartsng/assets/3985599/7779fda4-61fe-4fa2-8222-d9740f3e292e)

**Comment by quantenschaum** [comment]
OpenCPN also shows the white sector, but when the magnified sectors are displayed, you see the fainter white/yellow sector *and* the green sector.

![OpenCPN](https://github.com/wellenvogel/ochartsng/assets/3985599/f2362fe3-a411-4148-a8c7-33c284957f80)

**Comment by quantenschaum** [comment]
Same issue at Fehmarnsund bridge. The coloured sectors are completely covered by the fog light sector.

![image](https://github.com/wellenvogel/ochartsng/assets/3985599/8a014074-cb57-49cc-a607-9bce5f210ae7)

![image](https://github.com/wellenvogel/ochartsng/assets/3985599/0ac5c5f0-4743-41b8-a8fa-b0e11e0175b3)

**Comment by wellenvogel** [comment]
Thanks for bringing this up.
Unfortunately the radius handling for those lights is a bit chaotic - and as I took over the handling from OpenCPN...
Do you have a pointer to the style that is doing this at your site?
I did not really check if the handling is in line with the standard.

**Comment by quantenschaum** [comment]
I use [QGIS](https://www.qgis.org/) for the styling and rendering. You can design it interactively and the run in server mode and let MapProxy seed pull the tiles into the cache.

The instructions for the style are in https://github.com/quantenschaum/mapping/blob/master/bsh.qgs

- layer: BSH/aids to navigation
- style: sector light
- there are geometry generators for the sector boundaries and arcs
- global parameters are in the project variables (ctrl+shift-p)

**Comment by quantenschaum** [comment]
For my taste, the sector radii are generally too small compared to a classic (paper) chart to be able to read the sectors accurately.

I made the fog light arc dashed and added "(fog)" to the label to indicate, that it is exhibited during fog only.

**Comment by wellenvogel** [comment]
I agree on the size. Most probably I will add an option to scale the arcs and sectors independently from the symbol scale (currently they scale with the symbols).
Adding more text is always an issue of readability. And yes - dashed is nice - but requires more work in the code - and most probably more render time. So maybe I will come back to this a bit later.

**Comment by quantenschaum** [comment]
Nice. I made the radius proportional to the range of light and the map scale and clamped it to fixed limits. Fog lights get a factor of 0.7. Arcs with a radius below a threshold (20) are not shown at all.

from QGIS `clamp(0,60*if(valnmr,valnmr/30,1)*200000/@map_scale,50)*if(exclit=3,if(uband>3,0.7,0),1)` (units are QGIS millimeters)

- brighter light => bigger radius
- when zoomed in arcs have fixed size on screen
